packages/storage/amplify_storage_s3_dart/test/storage_s3_service/task/s3_download_task_test.dart

```diff
@@ -589,11 +599,16 @@ void main() {
                   throwsA(isA<StorageHttpStatusException>()),
                 );
               }
-            } else {
+            } else if (exception is NoSuchKey) {
               expect(
                 downloadTask.result,
                 throwsA(isA<StorageKeyNotFoundException>()),
               );
+            } else {
+              expect(
+                downloadTask.result,
+                throwsA(isA<NetworkException>()),
```

Comment:
So in this and similar tests are you expecting all 4 exceptions to be thrown?

StorageAccessDeniedException
StorageHttpStatusException
StorageKeyNotFoundException
NetworkException

Reply:
This test is driven by a for loop, each loop is an exception type to be tested.
They are not expected to be thrown at the same time.

Reply:
Okay that makes sense.  I've had similar approaches to unit tests in the past, but Jordan had a good point about keeping tests DAMP. 

Personally as an outside observer of your code, the use of for loops makes it hard for me to understand what you're testing.  I know it saves a lot of redundant code, but I don't think that's necessarily the point of writing unit tests.  It seems the file is already quite large, so perhaps it can be broken in to sub files if you're worried about file size. 

[HERE](https://enterprisecraftsmanship.com/posts/dry-damp-unit-tests/#:~:text=DRY%20stands%20for%20%22Don't,representation%20in%20your%20code%20base.&text=DAMP%20stands%20for%20%22Descriptive%20and,the%20readability%20of%20the%20code.)

Reply:
OK thanks for providing the reference, I think it makes sense.

I would argue however, this forEach driven test is DAMP as it's "for each exception type I'm doing this test"

I think the confusing point is the assertion part that is not somehow clear as the forEach itself. I will improve this part a bit to make it more clear.